pkg/connect/client/agent/v1alpha1/agent.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2025 Cofide Limited.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import (
+	"context"
+
+	agentpb "github.com/cofide/cofide-api-sdk/gen/go/proto/agent/v1alpha1"
+	agentsvcpb "github.com/cofide/cofide-api-sdk/gen/go/proto/connect/agent_service/v1alpha1"
+	federatedservicepb "github.com/cofide/cofide-api-sdk/gen/go/proto/federated_service/v1alpha1"
+	"github.com/spiffe/spire-api-sdk/proto/spire/api/types"
+	"google.golang.org/grpc"
+)
+
+// AgentClient is an interface for a gRPC client for the v1alpha1 version of the Connect AgentService.
+type AgentClient interface {
+	CreateAgentJoinToken(ctx context.Context, clusterID, trustZoneID string) (string, error)
+	UpdateTrustZoneBundle(ctx context.Context, bundle *types.Bundle) error
+	UpdateManagedTrustZoneBundle(ctx context.Context, trustZoneID string, bundle *types.Bundle) error
+	UpdateAgentStatus(ctx context.Context, status *agentpb.AgentStatus) error
+	// DEPRECATED: Federated services will move to a separate client.
+	RegisterFederatedService(ctx context.Context, fs *federatedservicepb.FederatedService) (string, error)
+	DeregisterFederatedService(ctx context.Context, federatedServiceID string) error
+	GetFederatedService(ctx context.Context, federatedServiceID string) (*federatedservicepb.FederatedService, error)
+	UpdateFederatedService(ctx context.Context, fs *federatedservicepb.FederatedService) error
+	ListFederatedServices(ctx context.Context) ([]*federatedservicepb.FederatedService, error)
+}
+
+type agentClient struct {
+	client agentsvcpb.AgentServiceClient
+}
+
+// New instantiates a new AgentClient for communication with a Connect API.
+func New(conn grpc.ClientConnInterface) AgentClient {
+	return &agentClient{
+		client: agentsvcpb.NewAgentServiceClient(conn),
+	}
+}
+
+func (c *agentClient) CreateAgentJoinToken(ctx context.Context, clusterID, trustZoneID string) (string, error) {
+	resp, err := c.client.CreateAgentJoinToken(ctx, &agentsvcpb.CreateAgentJoinTokenRequest{
+		ClusterId:   &clusterID,
+		TrustZoneId: &trustZoneID,
+	})
+	if err != nil {
+		return "", err
+	}
+	return resp.GetAgentToken(), nil
+}
+
+func (c *agentClient) UpdateTrustZoneBundle(ctx context.Context, bundle *types.Bundle) error {
+	trustzoneBundleUpdateRequest := &agentsvcpb.UpdateTrustZoneBundleRequest{
```

Comment:
```suggestion
	trustZoneBundleUpdateRequest := &agentsvcpb.UpdateTrustZoneBundleRequest{
```

Definitely more of a nit 😃

Reply:
Had to zoom in to see that diff!